### ISSUE TITLE
GraphQL Root category

### ISSUE THREAD

**Issue by bmxmale** [Smile-SA/elasticsuite#2098]
When we call graphql for root category:

```
{
  category(id: 1) {
    products {
      total_count
      page_info {
        current_page
        page_size
      }
    }
    children_count
  }
}
```
In results we get:

```
{
    "errors": [
        {
            "debugMessage": "Argument 1 passed to Smile\\ElasticsuiteCore\\Search\\Adapter\\Elasticsuite\\Request\\Query\\Builder::buildQuery() must implement interface Smile\\ElasticsuiteCore\\Search\\Request\\QueryInterface, null given",
            "message": "Internal server error",
            "extensions": {
                "category": "internal"
            },
            "locations": [
                {
                    "line": 3,
                    "column": 5
                }
            ],
            "path": [
                "category",
                "products"
            ]
        }
    ],
    "data": {
        "category": {
            "products": null,
            "children_count": "2"
        }
    }
}
```

Magento Version : CE 2.4.2

ElasticSuite Version : 2.10

Environment : Developer

**Comment by romainruaud** [comment]
I see that you are querying for the category "1" directly.

I'm not sure this should be supported considering the fact that this category is more a technical one than a "real" category.

We could fix this of course, but it might result in "no results" because in the Elasticsearch index, products are not attached to this particular "1" category, the lowest-level is the root category of the store view (which is usually "2").

**Comment by bmxmale** [comment]
I agree that is unusual query, but adding some statement avoid error displaying on query result.

**Comment by vahonc** [comment]
Hello Mateusz,

Yep, you're right and I'll try to do it.

BR,
Vadym

**Comment by rajnandinimai** [comment]
Is any updates on this, I am also facing same issue with both the categories i.e. for 1/2 and also this issue is when we pass 2(default category) in filter.

"errors": [
        {
            "debugMessage": "Smile\\ElasticsuiteCore\\Search\\Adapter\\Elasticsuite\\Request\\Query\\Builder::buildQuery(): Argument #1 ($query) must be of type Smile\\ElasticsuiteCore\\Search\\Request\\QueryInterface, null given",
            "message": "Internal server error",
            "extensions": {
                "category": "internal"
            },
            "locations": [
                {
                    "line": 1,
                    "column": 26
                }
            ],
            "path": [
                "products"
            ]
        }
    ]

**Comment by parthppwd** [comment]
Is there any fix? I am facing same issue in admin, when I go to Catalog > Categories and it loads default category it keep loading with below error in the `var/log/exception.log` 
`"exception":"[object] (TypeError(code: 0): Smile\\ElasticsuiteCore\\Search\\Adapter\\Elasticsuite\\Request\\Query\\Builder::buildQuery(): Argument #1 ($query) must be of type Smile\\ElasticsuiteCore\\Search\\Request\\QueryInterface, null given at /vendor/smile/elasticsuite/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Request/Query/Builder.php:50)"`
